### PR TITLE
RUM-4311: Add Logs event mapper to ObjC API

### PR DESCRIPTION
**Pull request by 0xnm**
### What and why?

This PR adds `setEventMapper` ObjC API for logs.

### How?

A brief description of implementation details of this PR.

### Review checklist
- [ ] Feature or bugfix MUST have appropriate tests (unit, integration)
- [ ] Make sure each commit and the PR mention the Issue number or JIRA reference
- [ ] Add CHANGELOG entry for user facing changes

### CODE REVIEW

File: DatadogObjc/Sources/Logs/LogsDataModels+objc.swift
Comment (ncreated) [inline DatadogObjc/Sources/Logs/LogsDataModels+objc.swift:1]: This looks 💯. How was it created? Long-term we may want to generate both Swift and Objc models through `rum-models-generator` CLI - I wonder how "far" is this file from the eventual result the generator may give us. It looks pretty solid and follows all the conventions from RUM models.

Comment (0xnm) [reply DatadogObjc/Sources/Logs/LogsDataModels+objc.swift:1]: It was created manually. I just took RUM models for the inspiration.

Comment (ncreated) [reply DatadogObjc/Sources/Logs/LogsDataModels+objc.swift:1]: Got it 👍. Sounds good as it solves current problem 👌. I hope we find time to code-gen this from JSON schema in `3.x`.

**Review by ncreated — APPROVED** [review]
Looks good 👌, great work on `+objc` models 👍.

**Pull request merged**